web_m2x_options/static/src/components/form.esm.js

```diff
@@ -0,0 +1,342 @@
+            props.canCreateEdit = false;
+        } else {
+            props.canCreateEdit = attrs.can_create
+                ? evaluateBooleanExpr(attrs.can_create)
+                : true;
+        }
+
+        return props;
+    },
+
+    m2oOptionsPropsLimit(props, attrs, options) {
+        const optionsLimit = Number(options.limit);
+        const irOptionsLimit = Number(session.web_m2x_options["web_m2x_options.limit"]);
+
+        props.searchLimit = optionsLimit || irOptionsLimit || props.searchLimit;
+
+        return props;
+    },
+
+    m2oOptionsPropsSearchMore(props, attrs, options) {
+        const irOptionsSearchMore =
+            session.web_m2x_options["web_m2x_options.search_more"];
+
+        if (options.search_more === false || irOptionsSearchMore === "False") {
+            props.noSearchMore = true;
+        } else if (options.search_more || irOptionsSearchMore === "True") {
+            props.noSearchMore = false;
+        }
+
+        return props;
+    },
+
+    m2oOptionsPropsOpen(props, attrs, options) {
+        const irOptionsOpen = session.web_m2x_options["web_m2x_options.open"];
+
+        if (options.open === false || irOptionsOpen === "False") {
+            props.canOpen = false;
+        } else if (options.open || irOptionsOpen === "True") {
+            props.canOpen = true;
+        }
+
+        return props;
+    },
+
+    m2oOptionsProps(props, attrs, options) {
+        const updatedProps = {
+            ...props,
+            ...this.m2oOptionsPropsCreate(props, attrs, options),
+            ...this.m2oOptionsPropsCreateEdit(props, attrs, options),
+            ...this.m2oOptionsPropsLimit(props, attrs, options),
+            ...this.m2oOptionsPropsSearchMore(props, attrs, options),
+            ...this.m2oOptionsPropsOpen(props, attrs, options),
+            fieldColor: options.field_color,
+            fieldColorOptions: options.colors,
+        };
+
+        return updatedProps;
+    },
+
+    extractProps({attrs, context, decorations, options, string}, dynamicInfo) {
+        const props = super.extractProps(
+            {attrs, context, decorations, options, string},
+            dynamicInfo
+        );
+        return this.m2oOptionsProps(props, attrs, options);
+    },
+});
+
+patch(Many2OneField.prototype, {
+    get Many2XAutocompleteProps() {
+        const {searchLimit, noSearchMore, fieldColor, fieldColorOptions} = this.props;
+        const props = super.Many2XAutocompleteProps;
+        const retProps = {...props};
+
+        if (Number(searchLimit) > 1) {
+            retProps.searchLimit = Number(searchLimit) - 1;
+        }
+
+        if (noSearchMore) {
+            retProps.noSearchMore = noSearchMore;
+        }
+
+        if (fieldColor && fieldColorOptions) {
+            retProps.fieldColor = fieldColor;
+            retProps.fieldColorOptions = fieldColorOptions;
+        }
+
+        return retProps;
+    },
+});
+
+patch(many2ManyTagsField, {
```

Comment:
are you missing `.prototype`, it should be `patch(many2ManyTagsField.prototype, {` right?

Reply:
we have [Many2OneField](https://github.com/odoo/odoo/blob/bd5d22eb24e80d8a8f65fa5535cf7fc78b80be0a/addons/web/static/src/views/fields/many2one/many2one_field.js#L59) class and [many2OneField](https://github.com/odoo/odoo/blob/bd5d22eb24e80d8a8f65fa5535cf7fc78b80be0a/addons/web/static/src/views/fields/many2one/many2one_field.js#L339C14-L339C27) object. so `many2ManyTagsField` is only patched itself